src/gsi/genqsat.f90

```diff
@@ -164,6 +162,7 @@ subroutine genqsat(qsat,tsen,prsl,lat2,lon2,nsig,ice,iderivative)
            qsat(i,j,k) = max(qmin,qsat(i,j,k))
 
            if(iderivative > 0)then
+!           if(es <= esmax .and. iderivative == 2 .and. qsat(i,j,k) > qmin )then
```

Comment:
Russ,

The line does not need to be retained.  However, I have retained it because it is not clear that we should be using the derivatives of moisture wrt t and p if the q value is reset to the minimum value (qmin).  Unfortunately it does appear to make some difference in the solution (not sure how big).

Reply:
Thank you, John, for explaining why this line is present.   If it's possible that we might use this line in the future it may be OK to retain it.  We can see what Mike says.